src/webgl/p5.RendererGL.js

```diff
@@ -1033,16 +1034,46 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       }
     }
 
-    // apply shader to pg
     pg.clear(); // prevent undesirable feedback effects accumulating secretly
-    pg.shader(this.filterShader);
-    this.filterShader.setUniform('tex0', this);
-    this.filterShader.setUniform('texelSize', [1.0/this.width, 1.0/this.height]);
-    this.filterShader.setUniform('canvasSize', [this.width, this.height]);
-    // filterParameter only used for POSTERIZE, BLUR, and THRESHOLD
-    // but shouldn't hurt to always set
-    this.filterShader.setUniform('filterParameter', filterParameter);
-    pg.rect(0,0,this.width,this.height);
+
+    // apply blur shader with multiple passes
+    if (operation === constants.BLUR) {
+
+      // pg is the accumulator. initialize with contents of main renderer (this)
+      pg.copy(
+        this,
+        0, 0, this.width, this.height,
+        -this.width/2, -this.height/2, this.width, this.height
+      );
+      // how much to blur, given by user
+      let steps = filterParameter;
+
+      for (let i = 0; i < steps; i++) {
+        // first pass averaging horizontal neighbors
+        pg.shader(this.filterShader);
+        this.filterShader.setUniform('texelSize', [1/this.width, 1/this.height]);
+        this.filterShader.setUniform('tex0', pg);
+        this.filterShader.setUniform('direction', [2, 0]); // 2 is a decent
+        pg.rect(0,0,this.width,this.height);               //  default spread
+        // another pass, this time vertically
+        pg.shader(this.filterShader);
+        // this.filterShader.setUniform('texelSize', [1/this.width, 1/this.height]);
+        this.filterShader.setUniform('tex0', pg);
```

Comment:
On the second pass, would it make sense to just draw directly to the main renderer?
```
// somewhere earlier we need to make the 2nd pass shader in the main renderer's context
this.filterShaderMain = new Shader(blur.vert, blur.frag);
// later when we're about to draw to screen
    this._pInst.push();
    this._pInst.shader(this.filterShaderMain);
    this.filterShaderMain.setUniform('texelSize', [1/this.width, 1/this.height]);
    this.filterShaderMain.setUniform('tex0', pg);
    this.filterShaderMain.setUniform('direction', [0, 2]);
    this._pInst.rect(0, 0, this.width, this.height);
    this._pInst.pop();
    ```

Reply:
That could work. I felt like it made more sense to do everything on the secondary renderer, and just leave the main renderer for the final step. Trying to keep things from being tangled together.